cumulusci/tasks/apex/testrunner.py

```diff
@@ -283,13 +303,81 @@ def _get_test_class_query(self):
         return query
 
     def _get_test_classes(self):
+        # If test_suite_names is provided, execute only tests that are a part of the list of test suites provided.
+        if (
+            self.options["test_suite_names"] is not None
+            and len(self.options["test_suite_names"]) > 0
+        ):
+            test_classes_from_test_suite_names = (
+                self._get_test_classes_from_test_suite_names()
+            )
+            return test_classes_from_test_suite_names
+
         query = self._get_test_class_query()
         # Run the query
         self.logger.info("Running query: {}".format(query))
         result = self.tooling.query_all(query)
         self.logger.info("Found {} test classes".format(result["totalSize"]))
         return result
 
+    def _get_comma_separated_string_of_items(self, itemlist):
+        # Accepts a list of strings. A formatted string is returned.
+        # Example: Input: ['TestSuite1', 'TestSuite2']      Output: ''TestSuite1','TestSuite2''
+        item_string = ""
+
+        for index, item in enumerate(itemlist):
+            item_string += f"'{item}'"
+            if index is not len(itemlist) - 1:
+                item_string += ","
+
+        return item_string
+
+    def _get_test_suite_ids_from_test_suite_names_query(self):
```

Comment:
This method gets easier to test if you take `test_suite_names_arg` as a parameter, instead of pulling it from the task's state.

Reply:
Thank you for pointing it out. This method as well as its respective tests has been updated as suggested.